### PR TITLE
[FIX] mail: channel/partner create error

### PR DESCRIPTION
**Pull request by niyasraphy**
**Description of the issue/feature this PR addresses:**
Error is raised on clicking the create button from the channel/member menu in settings.

1. Activate developer mode
2. Settings -> Technical -> Emails -> Channels/Member
3. Click on create button

**Current behavior before PR:**
Raising exception.

Screenshot from 2022-10-21 14-42-01

Desired behavior after PR is merged:
No exception

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by seb-odoo — CHANGES_REQUESTED** [review]
Hello,

Thank you for the report, I confirm the bug and it's ok to fix it, but did you try your code?

It's actually not working because the compute method is supposed to assign a value to every record, and it is currently skipping "new" (`NewId`) records with the current code.

`message_unread_counter` should be written to 0 for all members in `self` that don't have an actual id yet.

Also the `flush` should be kept next to the query, it is not necessary if the query is not done.

**Comment by niyasraphy** [comment]
@seb-odoo

File: addons/mail/views/mail_channel_member_views.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
         <field name="model">mail.channel.member</field>
         <field name="priority">10</field>
         <field name="arch" type="xml">
-            <tree string="Channels">
+            <tree string="Channels" create="0">
```
Comment (seb-odoo) [inline addons/mail/views/mail_channel_member_views.xml:9]: I would revert this diff part for now, I don't really see the need to change that in stable (it doesn't fix the bug).

In master, what could be done is to allow actually setting a value for channel_id when creating a new record, and only making it readonly when it is already set (+ adding sql constraint).

Comment (niyasraphy) [reply addons/mail/views/mail_channel_member_views.xml:9]: @seb-odoo thanks, should i remove from both the tree views or only this tree view ?

Comment (seb-odoo) [reply addons/mail/views/mail_channel_member_views.xml:9]: Just keep the change in the compute in this PR.

Comment (niyasraphy) [reply addons/mail/views/mail_channel_member_views.xml]: @seb-odoo updated the pr

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Comment by seb-odoo** [comment]
@niyasraphy there is an issue with ci/upgrade_enterprise, could you rebase your branch to hopefully fix it?

**Comment by niyasraphy** [comment]
@seb-odoo rebased

**Comment by seb-odoo** [comment]
@robodoo r+

**Pull request closed without merging**